### PR TITLE
Services must be public to be used in hooks

### PR DESCRIPTION
**Pull request by aschempp**
see https://github.com/contao/managed-edition/issues/38

### CODE REVIEW

File: src/Resources/contao/library/Contao/System.php [hunk 1]
```diff
@@ -164,6 +165,10 @@ protected function import($strClass, $strKey=null, $blnForce=false)
 			{
 				$this->arrObjects[$strKey] = $container->get($strClass);
 			}
+			elseif (strpos($strClass, '.') !== false)
+			{
+				throw new ServiceNotFoundException($strClass, null, null, [], sprintf('The service "%s" was not found or is not public. See https://symfony.com/doc/current/service_container.html#public-versus-private-services', $strClass));
```
Comment (leofeyer) [inline src/Resources/contao/library/Contao/System.php]: You also have to add `@throws ServiceNotFoundException` to the phpDoc.

File: src/DependencyInjection/Compiler/RegisterHookListenersPass.php [hunk 2]
```diff
@@ -54,6 +54,8 @@ private function getHooks(ContainerBuilder $container): array
             foreach ($tags as $attributes) {
                 $this->addHookCallback($hooks, $serviceId, $attributes);
             }
+
+            $container->findDefinition($serviceId)->setPublic(true);
```
Comment (leofeyer) [inline src/DependencyInjection/Compiler/RegisterHookListenersPass.php]: Are you aware that `findDefinition()` no longer includes aliases? We therefore check services and aliases separately in the [MakeServicesPublicPass](https://github.com/contao/core-bundle/blob/4.6/src/DependencyInjection/Compiler/MakeServicesPublicPass.php) class.

Comment (aschempp) [reply src/DependencyInjection/Compiler/RegisterHookListenersPass.php]: I am not aware of that… But can an alias be tagged and would it show up with the original service name or the alias name in `findTaggedServiceIds`?

Comment (leofeyer) [reply src/DependencyInjection/Compiler/RegisterHookListenersPass.php]: I don't know.

Comment (aschempp) [reply src/DependencyInjection/Compiler/RegisterHookListenersPass.php]: Fixed in a8fbc7f

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**